### PR TITLE
Update action to use node20

### PR DESCRIPTION
**Pull request by breathingdust**
Updates action.yml to use node20 and while we are here, perform minor dependency updates to resolve any vulnerabilities.

This also now imports nock-beta which includes experimental support for fetch. This was necessary to get the existing tests passing as mocked API calls were not being intercepted correctly. See: https://github.com/nock/nock/issues/2397

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+# v1.0.3
+
+NOTES
+
+- Set action to use node20
+- Update dependencies
+- Use nock-beta to allow mocking of HTTP requests in modern node. https://github.com/nock/nock/issues/2397
```
Comment (bflad) [inline CHANGELOG.md:7]:
```suggestion
# v2.0.0

NOTES

- Set action to use `node20`
- Update dependencies
```

Comment (bflad) [reply CHANGELOG.md:7]: We'll want to bump major version for non-hosted GitHub Actions runners. 👍 Also don't need to call out the testing change.

**Review by bflad — COMMENTED** [review]
Thank you for working on this, @breathingdust!

At the very least, we'll also need to bump the testing workflow in `.github/workflows/test.yml` to use `actions/reusable-workflows` commit `5e63e7aef083769958fc026fe8eea0241c0eb444` so it uses NodeJS 20 (or explicitly pass in `node-version: "20.x"`)

**Comment by bflad** [comment]
I also wouldn't be surprised if check-dist starts failing on NodeJS 20 -- running `npm run build` locally with NodeJS 20 should resolve that.

**Comment by bflad** [comment]
@breathingdust not sure why but rebuilding `dist` locally and updating jest in https://github.com/bflad/action-milestone-comment/pull/135 seems to resolve all the checks -- would you be okay if I pulled in that PR with your commits?

**Comment by breathingdust** [comment]
@bflad of course!

**Comment by bflad** [comment]
Going to pull in #135 and cut a release this afternoon 👍

**Comment by bflad** [comment]
Commits pulled in via #135

**Pull request closed without merging**